prow/npj/npj_test.go

```diff
@@ -206,8 +208,26 @@ func TestProwJobToPod(t *testing.T) {
 		// if !semantic.DeepEqual(got, test.expected) {
 		//	 t.Errorf("got pod:\n%#v\n\nexpected pod:\n%#v\n", got, test.expected)
 		// }
-		if _, ok := got.Metadata.Labels[kube.CreatedByProw]; !ok {
-			t.Error("expected a created-by-prow label")
+		var foundCreatedByLabel, foundTypeLabel, foundJobLabel bool
+		for key, value := range got.Metadata.Labels {
+			if key == kube.CreatedByProw && value == "true" {
+				foundCreatedByLabel = true
+			}
+			if key == "type" && value == string(pj.Spec.Type) {
+				foundTypeLabel = true
+			}
+			if key == "job" && value == pj.Spec.Job {
+				foundJobLabel = true
+			}
+		}
+		if !foundCreatedByLabel {
+			t.Error("expected a created-by-prow=true label")
```

Comment:
these errors should also show the actual labels. I would go with `t.Errorf("expected a created-by-prow=true label in %v", got.Metadata.Labels)`

Reply:
sure, updated